proto/dwtools/abase/l0.test/Long.test.s

```diff
@@ -11778,56 +11778,676 @@ function longAreRepeatedProbe( test )
 function longAllAreRepeated( test )
 {
 
-  var got = _.longAllAreRepeated( [] );
-  test.identical( got, true );
+  /* Without evaluator */
 
-  var got = _.longAllAreRepeated( [ 1, 1 ] );
-  test.identical( got, true );
+  test.case = 'empty array';
+  var src = [];
+  var got = _.longAllAreRepeated( src );
+  var expected = true;
+  test.identical( got, expected );
 
-  var got = _.longAllAreRepeated( [ 1 ] );
-  test.identical( got, false );
+  test.case = 'identical numbers of array';
+  var src = [ 1, 1 ];
+  var got = _.longAllAreRepeated( src );
+  var expected = true;
+  test.identical( got, expected );
 
-  var got = _.longAllAreRepeated( [ 1, 2, 2 ] );
-  test.identical( got, false );
+  test.case = 'identical strings of array';
+  var src = [ 'string', 'string' ];
+  var got = _.longAllAreRepeated( src );
+  var expected = true;
+  test.identical( got, expected );
+
+  test.case = 'single number of array';
+  var src = [ 1 ];
+  var got = _.longAllAreRepeated( src );
+  var expected = false;
+  test.identical( got, expected );
+
+  test.case = 'some numbers of array are identical';
+  var src = [ 1, 2, 2 ];
+  var got = _.longAllAreRepeated( src );
+  var expected = false;
+  test.identical( got, expected );
+
+  test.case = 'some strings of array are identical';
+  var src = [ 'string', 'string', 'diff string' ];
+  var got = _.longAllAreRepeated( src );
+  var expected = false;
+  test.identical( got, expected );
+
+  test.case = 'complex data of array';
+  var src = [ 1, 2, [ 3 ], 'string', 6, 6 ];
+  var got = _.longAllAreRepeated( src );
+  var expected = false;
+  test.identical( got, expected );
+
+  /**/
+
+  test.case = 'unroll empty';
+  var src = _.unrollMake( [] );
+  var got = _.longAllAreRepeated( src );
+  var expected = true;
+  test.identical( got, expected );
+
+  test.case = 'unroll, identical numbers';
+  var src = _.unrollMake( [ 1, 1 ] );
+  var got = _.longAllAreRepeated( src );
+  var expected = true;
+  test.identical( got, expected );
+
+  test.case = 'unroll, single number';
+  var src = _.unrollMake( [ 1 ] );
+  var got = _.longAllAreRepeated( src );
+  var expected = false;
+  test.identical( got, expected );
+
+  test.case = 'unroll, some numbers are identical';
+  var src = _.unrollMake( [ 1, 1, 2 ] );
+  var got = _.longAllAreRepeated( src );
+  var expected = false;
+  test.identical( got, expected );
+
+  test.case = 'unroll, complex data';
+  var src = _.unrollMake( [ 1, 2, [ 3 ], 'string', 6, 6 ] );
+  var got = _.longAllAreRepeated( src );
+  var expected = false;
+  test.identical( got, expected );
+
+  /* With evaluator */
+
+  test.case = 'empty array with evaluator';
+  var evaluator = ( e ) => e;
+  var src = [];
+  var got = _.longAllAreRepeated( src, evaluator );
+  var expected = true;
+  test.identical( got, expected );
+
+  test.case = 'single value of array with evaluator';
+  var evaluator = ( e ) => e.val;
```

Comment:
Good evaluator
```js
 var evaluator = ( e ) => e.val;
```
It returns some defined value. In this test case, it returns 1.

Please, note. Evaluator ( equalizer ) should not return `undefined` for most of elements. 

Example of wrong evaluator
```js
var evaluator = ( e ) => e[ 0 ];
var src = [ 1, 2, 3 ];
var got = _.longAllAreRepeated( src, evaluator );
```
For each element of `src` evaluator returns undefined.
So, use simple evaluator for such test cases. 
```js
var evaluator = ( e ) => e;
```